File: SDKs/googleapis.com/dfareporting/v4/python/src/sdk/models/shared/inventoryitem.py
from dataclasses import dataclass, field
from typing import Enum,List,Optional
from dataclasses_json import dataclass_json
from . import adslot
from . import lastmodifiedinfo
from . import pricing

class InventoryItemTypeEnum(str, Enum):
    PLANNING_PLACEMENT_TYPE_REGULAR = "PLANNING_PLACEMENT_TYPE_REGULAR"
    PLANNING_PLACEMENT_TYPE_CREDIT = "PLANNING_PLACEMENT_TYPE_CREDIT"


@dataclass_json
@dataclass
class InventoryItem:
    account_id: Optional[str] = field(default=None, metadata={'dataclasses_json': { 'field_name': 'accountId' }})
    ad_slots: Optional[List[adslot.AdSlot]] = field(default=None, metadata={'dataclasses_json': { 'field_name': 'adSlots' }})
    advertiser_id: Optional[str] = field(default=None, metadata={'dataclasses_json': { 'field_name': 'advertiserId' }})
    content_category_id: Optional[str] = field(default=None, metadata={'dataclasses_json': { 'field_name': 'contentCategoryId' }})
    estimated_click_through_rate: Optional[str] = field(default=None, metadata={'dataclasses_json': { 'field_name': 'estimatedClickThroughRate' }})
    estimated_conversion_rate: Optional[str] = field(default=None, metadata={'dataclasses_json': { 'field_name': 'estimatedConversionRate' }})
    id: Optional[str] = field(default=None, metadata={'dataclasses_json': { 'field_name': 'id' }})
    in_plan: Optional[bool] = field(default=None, metadata={'dataclasses_json': { 'field_name': 'inPlan' }})
    kind: Optional[str] = field(default=None, metadata={'dataclasses_json': { 'field_name': 'kind' }})
    last_modified_info: Optional[lastmodifiedinfo.LastModifiedInfo] = field(default=None, metadata={'dataclasses_json': { 'field_name': 'lastModifiedInfo' }})
    name: Optional[str] = field(default=None, metadata={'dataclasses_json': { 'field_name': 'name' }})
    negotiation_channel_id: Optional[str] = field(default=None, metadata={'dataclasses_json': { 'field_name': 'negotiationChannelId' }})
    order_id: Optional[str] = field(default=None, metadata={'dataclasses_json': { 'field_name': 'orderId' }})
    placement_strategy_id: Optional[str] = field(default=None, metadata={'dataclasses_json': { 'field_name': 'placementStrategyId' }})
    pricing: Optional[pricing.Pricing] = field(default=None, metadata={'dataclasses_json': { 'field_name': 'pricing' }})
    project_id: Optional[str] = field(default=None, metadata={'dataclasses_json': { 'field_name': 'projectId' }})
    rfp_id: Optional[str] = field(default=None, metadata={'dataclasses_json': { 'field_name': 'rfpId' }})
    site_id: Optional[str] = field(default=None, metadata={'dataclasses_json': { 'field_name': 'siteId' }})
    subaccount_id: Optional[str] = field(default=None, metadata={'dataclasses_json': { 'field_name': 'subaccountId' }})
    type: Optional[InventoryItemTypeEnum] = field(default=None, metadata={'dataclasses_json': { 'field_name': 'type' }})
    

File: SDKs/googleapis.com/dfareporting/v4/typescript/src/sdk/models/shared/inventoryitem.ts
import { SpeakeasyMetadata, SpeakeasyBase } from "../../../internal/utils";
import { AdSlot } from "./adslot";
import { LastModifiedInfo } from "./lastmodifiedinfo";
import { Pricing } from "./pricing";


export enum InventoryItemTypeEnum {
    PlanningPlacementTypeRegular = "PLANNING_PLACEMENT_TYPE_REGULAR",
    PlanningPlacementTypeCredit = "PLANNING_PLACEMENT_TYPE_CREDIT"
}


// InventoryItem
/** 
 * Represents a buy from the Planning inventory store.
**/
export class InventoryItem extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "json, name=accountId" })
  accountId?: string;

  @SpeakeasyMetadata({ data: "json, name=adSlots", elemType: AdSlot })
  adSlots?: AdSlot[];

  @SpeakeasyMetadata({ data: "json, name=advertiserId" })
  advertiserId?: string;

  @SpeakeasyMetadata({ data: "json, name=contentCategoryId" })
  contentCategoryId?: string;

  @SpeakeasyMetadata({ data: "json, name=estimatedClickThroughRate" })
  estimatedClickThroughRate?: string;

  @SpeakeasyMetadata({ data: "json, name=estimatedConversionRate" })
  estimatedConversionRate?: string;

  @SpeakeasyMetadata({ data: "json, name=id" })
  id?: string;

  @SpeakeasyMetadata({ data: "json, name=inPlan" })
  inPlan?: boolean;

  @SpeakeasyMetadata({ data: "json, name=kind" })
  kind?: string;

  @SpeakeasyMetadata({ data: "json, name=lastModifiedInfo" })
  lastModifiedInfo?: LastModifiedInfo;

  @SpeakeasyMetadata({ data: "json, name=name" })
  name?: string;

  @SpeakeasyMetadata({ data: "json, name=negotiationChannelId" })
  negotiationChannelId?: string;

  @SpeakeasyMetadata({ data: "json, name=orderId" })
  orderId?: string;

  @SpeakeasyMetadata({ data: "json, name=placementStrategyId" })
  placementStrategyId?: string;

  @SpeakeasyMetadata({ data: "json, name=pricing" })
  pricing?: Pricing;

  @SpeakeasyMetadata({ data: "json, name=projectId" })
  projectId?: string;

  @SpeakeasyMetadata({ data: "json, name=rfpId" })
  rfpId?: string;

  @SpeakeasyMetadata({ data: "json, name=siteId" })
  siteId?: string;

  @SpeakeasyMetadata({ data: "json, name=subaccountId" })
  subaccountId?: string;

  @SpeakeasyMetadata({ data: "json, name=type" })
  type?: InventoryItemTypeEnum;
}
